src/mlpack/core/util/size_checks.hpp

```diff
@@ -0,0 +1,118 @@
+/**
+ * @file size_checks.hpp
+ * @author Kirill Mishchenko
+ * @author Bisakh Mondal
+ *
+ * Utility for checking same size & same dimensionality.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_UTIL_SIZE_CHECKS_HPP
+#define MLPACK_UTIL_SIZE_CHECKS_HPP
+
+#include <mlpack/core.hpp>
```

Comment:
```suggestion
```

I think this is the cause of the build issue.  Since we are including `size_checks.hpp` inside of `prereqs.hpp`, we don't want to include `core.hpp` here.  In fact, by the time we include this file, it will be included from `prereqs.hpp` and everything else we need for this file should be included already.  So I think we can just remove this line. :+1: